### DISCUSSION TITLE
Referencing module functions in `typealias` type constraints?

### DISCUSSION THREAD

**Discussion by James-Ansley** [apple/pkl#208] [Q&A]
I am playing around with a package to validate strings against the New Zealand NHI standard. Ideally, I would like an importable type alias that encapsulates the validation logic.

Within one module (`nhi.pkl`), I have the type alias `typealias Nhi = ...` and a function `isValid` that checks if a string satisfies the standard. However, when I try to do:

```pkl
typealias Nhi = String(isValid(this))
```

I get the error ``Cannot find method `isValid`.`` when I try to use this type alias in another module

However, if I dynamically import the nhi module (from within itself), it works:

```pkl
typealias Nhi = String(import("nhi.pkl").isValid(this))
```

Based on #144 it seems type aliases are just inlined wherever they're used. But, Is this the intended way to create importable type aliases that reference other functions within a module? Not sure if I'm missing something in the docs.

**Comment by holzensp** [comment]
This reads like a straightforward name resolution issue, which #144 should indeed fix for you. Type aliases were being inlined _entirely_, without resolving names used in the constraints _in their definition scope_, leaving them to be resolved in their _use site scope_. Can you try the post-#144 version on your Pkl code?
